packages/layout/src/tree-layout/layout-definition/helper/index.js

```diff
@@ -0,0 +1,290 @@
+
+            left: instance.parentCut() === 'h'
+        ? lastSibling.boundBox().left : lastSibling.boundBox().left + lastSibling.boundBox().width
+        };
+    }
+  // if first sibling, take boundbox from parent
+    return {
+        width: bb.width,
+        height: bb.height,
+        top: instance.parent().boundBox().top,
+        left: instance.parent().boundBox().left
+    };
+}
+
+// prepares the targetComponent Map for target Mapping ie. where a component should lie
+export function prepareTargetComponentMap (context) {
+    context.targetComponentMap(new Map());
+    context.componentMap().forEach((value) => {
+        if (context.targetComponentMap().has(value.target())) {
+            context.targetComponentMap().get(value.target()).push(value);
+        } else {
+            const temp = [];
+            temp.push(value);
+            context.targetComponentMap().set(value.target(), temp);
+        }
+    });
+}
+
+export function getComponent (canvasComponent, componentName) {
+    const comp = canvasComponent.find(component => component.name() === componentName);
+    return (comp && comp !== -1) ? comp : null;
+}
+export function createPlaceHolderComponent (height, width, position) {
+    const comp = new DummyComponent(0, { height, width });
+    comp.name('placeHolder');
+    comp.position(position);
+    return comp;
+}
+
+export function placeComponent (definitionModel, component, isPreferred = false, isGridComponent = false) {
+    if (!component) {
+        return { first: definitionModel, second: definitionModel };
+    }
+    const componentDimension = component.getLogicalSpace();
+    const componentHeight = componentDimension.height;
+    const componentWidth = componentDimension.width;
+    let cut = '';
+    let componentRatioWidth = 1;
+    let leftOvercomponentRationWidth = 1;
+    let leftHeight = 0;
+    let leftWidth = 0;
+    if (component.position() === 'top' || component.position() === 'bottom') {
+        cut = 'h';
+        componentRatioWidth = componentHeight / definitionModel.remainingHeight();
+        leftHeight = definitionModel.remainingHeight() - componentHeight;
+        leftWidth = definitionModel.remainingWidth();
+    } else {
+        cut = 'v';
+        componentRatioWidth = componentWidth / definitionModel.remainingWidth();
+        leftWidth = definitionModel.remainingWidth() - componentWidth;
+        leftHeight = definitionModel.remainingHeight();
+    }
+    leftOvercomponentRationWidth = 1 - componentRatioWidth;
+
+// update parentModel
+    definitionModel.cut(cut);
+
+    const firstLane = new DefinitionModel(component.name(),
+                                            null,
+                                            componentRatioWidth,
+                                            isGridComponent ? false : isPreferred,
+                                            []);
+    firstLane.remainingHeight(componentHeight);
```

Comment:
Can firstLane.remainingHeight be chained? IF not, make it chainable
(not necessary to use it as a chained call, but the capability shud exist)